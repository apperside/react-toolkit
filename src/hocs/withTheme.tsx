import React from 'react';
import { baseTheme, ThemeProvider } from '../ui/theme';

export function withTheme<P>(WrappedComponent: React.ComponentType<P>): React.FC<P> {
    const ComponentWithTheme = (props: P) => {
        return (
            <ThemeProvider theme={baseTheme as any}>
                <WrappedComponent {...props} />
            </ThemeProvider>
        );
    };
    return ComponentWithTheme;
}
export default withTheme;
