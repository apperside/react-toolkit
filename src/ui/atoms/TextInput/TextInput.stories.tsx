import React from 'react';
import { Story, Meta } from '@storybook/react';

import TextInput from './TextInput';
import { TextInputProps } from "./TextInput.styled";

export default {
    title: 'Components/TextInput',
    component: TextInput,
    argTypes: {
        backgroundColor: { control: 'color' },
        size: {
            control: {
                type: 'select',
                options: ['small', 'medium', 'large'],
            },
        },
        borderRadius: {
            control: {
                type: 'select',
                options: [2, 4, 8],
            },
        },
    },
} as Meta;

const Template: Story<TextInputProps> = (args) => <TextInput {...args} >Hello world</TextInput>;

export const Primary = Template.bind({});
Primary.args = {
    variant: 'primary',
    // label: 'TextInput',
};

export const Secondary = Template.bind({});
Secondary.args = {
    variant: 'secondary',
    // label: 'TextInput',
};
