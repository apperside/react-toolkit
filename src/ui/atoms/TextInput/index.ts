export * from "./TextInput"
export * from "./TextInput.styled"
export * from "./TextInput.theme"