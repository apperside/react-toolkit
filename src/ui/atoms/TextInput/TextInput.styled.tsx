import shouldForwardProp from "@styled-system/should-forward-prop";
import { PropsOf } from "../../../typeUtils";
import styled from 'styled-components';
import { FlexboxProps } from "styled-system";
import { background, BackgroundImageProps, BorderProps, borderRadius, TextStyleProps as SSTextStyleProps, color, ColorProps, flexbox, fontFamily, FontFamilyProps, fontSize, fontStyle, fontWeight, layout, LayoutProps, letterSpacing, lineHeight, size, space, SpaceProps, textAlign, textStyle, TextStyleProps, TypographyProps as SSTypographyProps, variant, width, WidthProps } from "styled-system";
import { PropsWithVariant } from "../../theme";
import { TextInputVariants } from "./TextInput.theme";

export type TextInputStyleProps =
    & SSTypographyProps
    & SpaceProps
    & WidthProps
    & ColorProps
    & BackgroundImageProps
    & SSTextStyleProps
    & LayoutProps
    & FlexboxProps
    & FontFamilyProps
    & BorderProps

export interface TextInputProps extends TextInputStyleProps, PropsWithVariant<keyof TextInputVariants> {
    // children: React.ReactNode;
    value?: string | number
    onEnterPress?: () => void
    // onClick?: () => void
}
const InternalThemedTextInput = styled.input.withConfig<TextInputProps>({
    // avoid forwarding styled-system's props to dom
    shouldForwardProp: (prop: any) => prop === "placeholderTextColor" || prop === "keyboardType" || shouldForwardProp(prop)

})<TextInputProps>(
    space,
    fontSize,
    fontStyle,
    size,
    color,
    textStyle,
    letterSpacing,
    fontFamily,
    fontWeight,
    borderRadius,
    lineHeight,
    textAlign,
    background,
    flexbox,
    layout,
    width,
    props => variant({
        variants: props.theme.textInput.variants as any
    }),
    {
        boxSizing: "border-box"
    }
);

type ThemedTextInputProps = Omit<PropsOf<typeof InternalThemedTextInput>, "size" | "width" | "height"> & LayoutProps;

export const ThemedTextInput: React.FC<ThemedTextInputProps> = InternalThemedTextInput;