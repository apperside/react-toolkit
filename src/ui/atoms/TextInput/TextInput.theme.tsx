import { VariantProps } from "../../theme";
import { TextInputStyleProps } from "./TextInput.styled";


// type BaseTextInputVariants = { [key: string]: VariantProps<TextInputProps> }

export interface TextInputVariants {
  primary: VariantProps<TextInputStyleProps>;
  secondary: VariantProps<TextInputStyleProps>;
}

/**
 * The new type for a themed component with variants
 * @param Props type of component's props
 * @param Variant the type defining the component's variants
 * 
 * EG: type MyComponentVariants = "primary" | "secondary"
 */
// export type ThemedComponentWithVariants<Props, Variants extends string> =
//   Props & { variants: VariantsProps<Props, Variants> }

export type TextInputTheme = TextInputStyleProps & { variants: Partial<TextInputVariants> }
// interface Variants<T>={[key:keyof:T]}
export const textInputTheme: TextInputTheme = {
  fontFamily: "body",
  py: 10, px: 15,
  borderStyle: "solid",
  borderWidth: 1,

  variants: {
    primary: {
      borderColor: "primary",
    },
    secondary: {
      borderColor: "secondary"
    },
  }
};