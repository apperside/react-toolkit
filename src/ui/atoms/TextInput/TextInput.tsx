import React from 'react';
import { PropsOf } from "src/typeUtils";
import { TextInputProps, ThemedTextInput } from './TextInput.styled';

export const TextInput: React.FC<PropsOf<typeof ThemedTextInput>> = ({
    variant = "primary",
    ...props
}) => {
    return (
        <ThemedTextInput variant={variant} value="ciao" {...props} />
    )
};

export default TextInput;
