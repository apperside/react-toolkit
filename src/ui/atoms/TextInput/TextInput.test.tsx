import { render } from '@testing-library/react';
import React from 'react';
import withTheme from "src/hocs/withTheme";
import TextInput from './TextInput';

const ThemedTextInput = withTheme(TextInput);
describe('Text', () => {
    test('renders a default button with text', () => {
        const { baseElement } = render(<ThemedTextInput readOnly value="Hello world" />);

        // expect(screen.getByText('Hello world')).toBeInTheDocument();
        // expect(screen.getByText('Click me')).toHaveStyle({
        //     backgroundColor: '#025D92',
        //     color: 'white',
        // });
        expect(baseElement).toMatchSnapshot();
    });
    // test('renders a primary button', () => {
    //     const { baseElement } = render(<ThemedText variant="primary" label={'Click me'} />);

    //     expect(screen.getByText('Click me')).toHaveStyle({
    //         backgroundColor: '#025D92',
    //         color: 'white',
    //     });
    //     expect(baseElement).toMatchSnapshot();
    // });
    // test('handles onClick', () => {
    //     const mockOnClick = jest.fn();
    //     const { baseElement } = render(<ThemedText label={'Click Me'} onClick={mockOnClick} />);
    //     userEvent.click(screen.getByText('Click Me'));

    //     expect(mockOnClick).toHaveBeenCalledTimes(1);
    //     expect(baseElement).toMatchSnapshot();
    // });
});
