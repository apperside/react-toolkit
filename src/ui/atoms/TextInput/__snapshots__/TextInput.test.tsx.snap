// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders a default button with text 1`] = `
<body>
  <div>
     
    <input
      class="sc-bdnxRM zUpI"
      readonly=""
      value="Hello world"
    />
     
  </div>
</body>
`;
